kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/Tab8021xUi.java

```diff
@@ -0,0 +1,350 @@
+    }
+
+    private void initUsernameTextBox() {
+        this.username.addMouseOverHandler(event -> {
+            Tab8021xUi.this.logger.info("hover detected.");
+        });
+        this.username.addBlurHandler(e -> this.username.validate());
+        this.username.setAllowBlank(true);
+        this.username.addMouseOutHandler(event -> resetHelpText());
+
+        this.username.addChangeHandler(event -> {
+            setDirty(true);
+        });
+    }
+
+    private void initPasswordTextBox() {
+        this.password.addMouseOverHandler(event -> {
+            Tab8021xUi.this.logger.info("hover detected.");
+        });
+
+        this.password.addBlurHandler(e -> this.password.validate());
+        this.password.setAllowBlank(true);
+        this.password.addMouseOutHandler(event -> resetHelpText());
+
+        this.password.addChangeHandler(event -> {
+            setDirty(true);
+        });
+    }
+
+    private void reset() {
+        for (int i = 0; i < this.eap.getItemCount(); i++) {
+            if (this.eap.getSelectedItemText().equals(Gwt8021xEap.TTLS.name())) {
+                this.eap.setSelectedIndex(i);
+                break;
+            }
+        }
+
+        for (int i = 0; i < this.innerAuth.getItemCount(); i++) {
+            if (this.innerAuth.getSelectedItemText().equals(Gwt8021xInnerAuth.NONE.name())) {
+                this.innerAuth.setSelectedIndex(i);
+                break;
+            }
+        }
+
+        this.password.setText("");
+        this.username.setText("");
+
+        update();
+    }
+
+    private void setValues() {
+        logger.info("Start setValues");
+
+        if (this.activeConfig == null) {
+            // return;
+        }
+
+        for (int i = 0; i < this.eap.getItemCount(); i++) {
+            if (this.eap.getItemText(i).equals(this.activeConfig.getEapEnum().name())) {
+                this.eap.setSelectedIndex(i);
+                break;
+            }
+        }
+
+        for (int i = 0; i < this.innerAuth.getItemCount(); i++) {
+            if (this.innerAuth.getItemText(i).equals(this.activeConfig.getInnerAuthEnum().name())) {
+                this.innerAuth.setSelectedIndex(i);
+                break;
+            }
+        }
+
+        this.username.setValue(GwtSafeHtmlUtils.htmlUnescape(this.activeConfig.getUsername()));
+        this.password.setValue(GwtSafeHtmlUtils.htmlUnescape(this.activeConfig.getPassword()));
+    }
+
+    private void update() {
+        setValues();
+    }
+
+    @Override
+    public void clear() {
+        // Not needed
+    }
+
+    @Override
+    public void refresh() {
+        if (isDirty()) {
+            setDirty(false);
+            if (this.activeConfig == null) {
+                reset();
+            } else {
+                update();
+            }
+        }
+    }
+
+    @Override
+    public boolean isDirty() {
+        return this.dirty;
+    }
+
+    @Override
+    public boolean isValid() {
+        return true;
+    }
```

Comment:
We are allowing to set an empty password when filling the username. This leads to the following error in the `kura.log`:

```
java.util.NoSuchElementException: The "net.interface.wlan0.config.802-1x.password" key contains an empty string value.
	at org.eclipse.kura.nm.NetworkProperties.get(NetworkProperties.java:60)
	at org.eclipse.kura.nm.configuration.NMSettingsConverter.create8021xMschapV2(NMSettingsConverter.java:208)
	at org.eclipse.kura.nm.configuration.NMSettingsConverter.build8021xSettings(NMSettingsConverter.java:140)
	at org.eclipse.kura.nm.configuration.NMSettingsConverter.buildSettings(NMSettingsConverter.java:97)
	at org.eclipse.kura.nm.NMDbusConnector.enableInterface(NMDbusConnector.java:443)
	at org.eclipse.kura.nm.NMDbusConnector.manageConfiguredInterface(NMDbusConnector.java:422)
	at org.eclipse.kura.nm.NMDbusConnector.doApply(NMDbusConnector.java:381)
	at org.eclipse.kura.nm.NMDbusConnector.lambda$0(NMDbusConnector.java:366)
	at java.util.ArrayList.forEach(ArrayList.java:1259)
	at org.eclipse.kura.nm.NMDbusConnector.doApply(NMDbusConnector.java:363)
	at org.eclipse.kura.nm.NMDbusConnector.apply(NMDbusConnector.java:322)
	at org.eclipse.kura.nm.configuration.NMConfigurationServiceImpl.writeNetworkConfigurationSettings(NMConfigurationServiceImpl.java:317)
	at org.eclipse.kura.nm.configuration.NMConfigurationServiceImpl.update(NMConfigurationServiceImpl.java:218)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.felix.scr.impl.inject.methods.BaseMethod.invokeMethod(BaseMethod.java:228)
	at org.apache.felix.scr.impl.inject.methods.BaseMethod.access$500(BaseMethod.java:41)
	at org.apache.felix.scr.impl.inject.methods.BaseMethod$Resolved.invoke(BaseMethod.java:664)
	at org.apache.felix.scr.impl.inject.methods.BaseMethod.invoke(BaseMethod.java:510)
	at org.apache.felix.scr.impl.inject.methods.ActivateMethod.invoke(ActivateMethod.java:317)
	at org.apache.felix.scr.impl.inject.methods.ActivateMethod.invoke(ActivateMethod.java:307)
	at org.apache.felix.scr.impl.manager.SingleComponentManager.invokeModifiedMethod(SingleComponentManager.java:836)
	at org.apache.felix.scr.impl.manager.SingleComponentManager.modify(SingleComponentManager.java:791)
	at org.apache.felix.scr.impl.manager.SingleComponentManager.reconfigure(SingleComponentManager.java:709)
	at org.apache.felix.scr.impl.manager.SingleComponentManager.reconfigure(SingleComponentManager.java:673)
	at org.apache.felix.scr.impl.manager.ConfigurableComponentHolder.configurationUpdated(ConfigurableComponentHolder.java:435)
	at org.apache.felix.scr.impl.manager.RegionConfigurationSupport.configurationEvent(RegionConfigurationSupport.java:316)
	at org.apache.felix.scr.impl.manager.RegionConfigurationSupport$2.configurationEvent(RegionConfigurationSupport.java:118)
	at org.eclipse.equinox.internal.cm.EventDispatcher$1.run(EventDispatcher.java:92)
	at org.eclipse.equinox.internal.cm.SerializedTaskQueue$1.run(SerializedTaskQueue.java:40)
```

We should check that all the required fields are ok before submitting the form. This is usually done in the `Tab8021xUi.java::isValid` which now always returns true.